### ISSUE TITLE
Add CSS escape hatches

### ISSUE THREAD

**Issue by patricklafrance** [retypeapp/retype#639]
I would like to propose the introduction of CSS escape hatches to Retype. I believe that could benefit not only myself but also potentially a wider range of developers and use cases as layout preferences tend to be very subjective. 

By adding CSS escape hatches, that would enable developers to customize a few parts of Retype's layouts using CSS and would alleviate the pressure on the Retype core team to work on layout issues until they are ready to do so. Of course, that would be at the own risk of the developers who choose to leverage the CSS escape hatches.

In my experience, it's pretty low effort solution to put in place and doesn't negatively impact the maintenance of the project.

Personally, I would like Retype to provide CSS escape hatches so I can hide a few elements of the mobile layout.

There are examples of other platforms using CSS escape hatches to allow this type of customization, one of those is [Storybook](https://storybook.js.org/docs/react/configure/theming#css-escape-hatches).

What is exactly CSS escape hatches? It's giving the ability to the developers to target specific elements that are rendered by Retype by adding a unique CSS class or ID to those elements.

Here's a few elements that I would like to have CSS escape hatches for.

Desktop layout:

![css_escape_hatches_2](https://github.com/retypeapp/retype/assets/794579/a123445f-3f03-483e-b996-11de9e758f14)

Mobile layout:

![css_escape_hatches_3](https://github.com/retypeapp/retype/assets/794579/6b26de34-db5e-4bce-9805-5a81751a2c99)

Mobile layout, menu opened:

![css_escape_hatches_4](https://github.com/retypeapp/retype/assets/794579/d4ed4d44-340a-4af6-a1c8-1bb8dcfcb300)

Let me know what you think!

Thank you,

Patrick

**Comment by geoffreymcgill** [comment]
Yes, this is definitely something we can do. It's a relatively easy task. 

Some of these elements already have id's or unique CSS classes applied, although not consistently.

What is your opinion on namespacing the `id` values with a `retype-` prefix? For example, using `retype-header-nav` vs `header-nav`. 

The `retype-` prefix would help avoid id collisions with other custom elements that authors or plugins might add to the HTML, but the downside is a longer `id` value. Collisions without namespacing may be rare, although `id` values are cleaner when they follow a predictable generic simplified naming convention.

**Comment by patricklafrance** [comment]
I agree, prefixing the ids with `retype-` is a great idea and a pattern that we usually also implements for our libraries 👍🏻

**Comment by geoffreymcgill** [comment]
Here is a list of unique new element `id` values that have been added to the generated HTML elements. 

ID | Description
--- | ---
`retype-app` | Root application container
`retype-header` | Head section container
`retype-header-nav` | Main navigation in header
`retype-branding-logo` | Custom branding logo
`retype-container` | Main container for content and navigation
`retype-content` | Main content region
`retype-content-footer` | Page content footer
`retype-breadcrumb` | Breadcrumb container
`retype-sidebar-left` | Left sidebar container
`retype-sidebar-right` | Right sidebar container
`retype-nextprev` | Container for `nextprev` page navigation component
`retype-edit-this-page` | "Edit this page" link
`retype-tags` | Tags container
`retype-copyright` | Copyright container
`retype-footer-links` | Footer links
`retype-sidebar-filter` | Sidebar filter container
`retype-search` | Sidebar search container
`retype-mobile-search-button` | Mobile search button
`retype-scheme-button` | Scheme switcher button
`retype-history-button` | History button
`retype-sidebar-left-toggle-button` | Left sidebar toggle button
`retype-sidebar-right-toggle-button` | Right sidebar toggle button
`retype-back-to-top` | Back to top button
`retype-page-footer` | Main page footer

> [!warning]
> Some existing `id` values that were available in previous releases have been renamed from `docs-` prefix to `retype-`

If you find an element that is missing an `id` and feel it should have an `id` assigned, just let us know and we will add. These are relatively easy for us to add at any time, so there should be a quick turn-around on getting the `id` added and included in the next release. 

The `id` values above will be included in the upcoming `v3.10.0` release.

Hope this helps.
